be/src/olap/olap_server.cpp

```diff
@@ -876,9 +877,12 @@ std::vector<TabletSharedPtr> StorageEngine::_generate_compaction_tasks(
 }
 
 void StorageEngine::_update_cumulative_compaction_policy() {
-    if (_cumulative_compaction_policy == nullptr) {
-        _cumulative_compaction_policy =
-                CumulativeCompactionPolicyFactory::create_cumulative_compaction_policy();
+    if (_all_cumulative_compaction_policy.size() < 2) {
```

Comment:
check size 2 is tricky. check _all_cumulative_compaction_policy.count(policyxx).

BTW, name _cumulative_compaction_policies is better

Reply:
done